string init_inputs_outputs
---
string current_inputs_outputs